### DISCUSSION TITLE
Auto-trigger the insert template modal?

### DISCUSSION THREAD

**Discussion by Fierce-Ninja** [SilentVoid13/Templater#410] [Help]
Hi there.  Is there a way to auto-trigger the "Open Insert template modal" command of the templater plugin when creating a new note?  So that the command palette opens up with the drop down of all existing templates.

**Comment by ccc-dvansevenant** [comment]
You could remap the "Create new note" hotkey, or designate a new hotkey, assigning to "Templater: Create new note from template"

![image](https://user-images.githubusercontent.com/41758088/137798456-0dd2754b-ad64-4cdb-9987-d3e22269d11c.png)

When triggered, it asks which template you want to use:

![image](https://user-images.githubusercontent.com/41758088/137798551-225e249d-6c4d-416c-bdb9-6df82648faf3.png)

Is that something like what you are trying to do?

**Reply by Fierce-Ninja** [reply]
That would work.  Except that I am not good at writing Templater commands 🥺 Any help? Or if it is too much, then please point me to the best resources. Not too savvy but I can hopefully pick up

**Reply by ccc-dvansevenant** [reply]
Start small. Check the Obsidian Forum, or the Templater Discussion here on GitHub, or search YouTube.

An example of a basic template might be like this:

```
<< [[<% tp.date.now("YYYY-MM-DD", -1) %>]] | [[<% tp.date.now("YYYY-MM-DD", 1) %>]] >>

# <% tp.file.title %> Daily Note

## TODO
<% tp.file.cursor(0) %>


## Notes


## Meetings
```

The Templater docs do list things out, but are not easy to navigate, I find.

**Comment by Fierce-Ninja** [comment]
One issue I found with the above suggestion is that the templater doesn't update the tp.file.title after the fact.  Meaning, let's say I use a hotkey and have the templater directly open a new note with the template as suggested above.  Let's say in that template I have <%tp.file.title%>.  The templater will insert something like "untitled ##".  So even if I retitle my note, which I would be doing for the first time since it is a brand new note, the templater-populated note would continue to show "untitled ##" as the note title in the note.  Is there a way I get a chance to title the note and the templater can then update the tp.file.title in the note itself?

**Reply by melbv** [reply]
Hi there
I had the same issue as you as wanted to have an automated way to create a note & have templater pre-fill in according to a specific template. 
below is the solution i came to:
In your template folder, create a note that will generate your note for you (I called it 'NewFile').
Paste the below code:
```
<%*
const fileName = await tp.system.prompt("File name")
const templateName = await tp.system.suggester(["Template1", "Template2"], ["Template1", "Template2"])
tp.file.create_new(tp.file.find_tfile(templateName), fileName, tp.file.folder())
%>
```

replace the sample table items "Template1" & "Template2" with the name of your templates.

By calling this template with code ('NewFile'), a freetext box will appear to enter your note's name (NOT title) and a drop-down will then appear to let you choose your template from the list you have parametered. I personnally use notes' frontmatter with a field 'ChildrenType' to predefine which templates to pre-populate as i use that script for all my master notes....

For you to then create a note, either call the 'NewFile' file from the command palette (create a note from a template) or create a button somewhere (with the Buttons plugin):

```
\```button
name Create Note
type append template
action NewFile
\```
```

Dont hesitate to ask if anything is unclear

**Reply by ccc-dvansevenant** [reply]
Brilliant! I'm going to have to try this. 

Ran in to the same issue: Nice templates, but starting with them default to "Untitled". Or, have to create the file first, then insert the template.

The little script @melbv suggests may just do the trick. Name it *and* pick a template!

**Comment by melbv** [comment]
It is working for me and i am sure it willfor you too! Dont heaitate to reach out if it does not

25 Oct 2021, 19:26 by ***@***.***:
